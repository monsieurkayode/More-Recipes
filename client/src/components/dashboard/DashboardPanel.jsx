import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'proptypes';
import Avatar from 'react-avatar';

import dash from '../../../assets/css/img/dash.jpg';

/**
 * DashboardPanel
 * @function DashboardPanel
 *
 * @param {object} props
 *
 * @return {JSX} JSX
 */
const DashboardPanel = props => (
  <div
    id="dash-panel"
    className="col l3 teal hide-on-med-and-down"
  >
    <div id="profile-page-header" className="card teal z-depth-0">
      <div className="card-image">
        <img
          className="responsive-img"
          id="dash-img"
          src={dash}
          alt="user-background"
        />
        <div className="card-profile-image">
          <Avatar
            className="card-profile-avatar"
            name={props.user.username}
            size={80}
            color="#345"
            round
          />
        </div>
      </div>
      <div id="dash-links" >
        <div className="collection">
          <a
            id="dashboard-parent"
            className="collection-item teal"
          >
            <strong>
              <span className="fa fa-dashboard" /> Dashboard
            </strong>
          </a>
          <br />
          <div id="myTab" className="tabs-vertical">
            <ul className="tabs teal">
              <li className="tab">
                <Link
                  className={`white-text 
                    ${props.route === 'profile' && 'active'}`
                  }
                  to="/dashboard/profile"
                >
                  <i className="fa fa-vcard-o" /> My Profile
                </Link>
              </li>
              <li className="tab">
                <Link
                  to="/dashboard/recipes"
                  className={`white-text
                    ${props.route === 'recipes' && 'active'}`
                  }
                >
                  <i className="fa fa-briefcase" /> My Recipes
                </Link>
              </li>
              <li className="tab">
                <Link
                  to="/dashboard/favorites"
                  className={`white-text 
                    ${props.route === 'favorites' && 'active'}`
                  }
                >
                  <i className="fa fa-heart" /> Favorite Recipes
                </Link>
              </li>
              <li className="tab">
                <Link
                  to="/dashboard/notifications"
                  className={`white-text 
                    ${props.route === 'notifications' && 'active'}`
                  }
                >
                  <i className="fa fa-comments-o" /> Notifications
                  <span id="not-badge" className="red new badge">10</span>
                </Link>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>
);

DashboardPanel.defaultProps = {
  user: {}
};

DashboardPanel.propTypes = {
  user: PropTypes.shape({
    username: PropTypes.string
  }),
  route: PropTypes.string.isRequired
};

export default DashboardPanel;
