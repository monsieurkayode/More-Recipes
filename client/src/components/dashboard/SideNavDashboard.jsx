import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'proptypes';
import Avatar from 'react-avatar';

import dash from '../../../assets/css/img/dash.jpg';

/**
 * SideNavDashboard
 * @function SideNavDashboard
 *
 * @param {object} props
 *
 * @return {JSX} JSX
 */
const SideNavDashboard = props => (
  <div className="side-nav" id="dashboard">
    <div id="dash-panel" className="col l3 teal">
      <div className="teal">
        <div
          id="profile-page-header"
          className="card teal z-depth-0"
        >
          <div className="card-image">
            <img
              className="responsive-img"
              id="dash-img"
              src={dash}
              alt="user-background"
            />
            <div className="card-profile-image">
              <Avatar
                className="card-profile-avatar responsive-img"
                name={props.user.username}
                size={80}
                color="#345"
                round
              />
            </div>
          </div>
          <div id="dash-links">
            <div className="collection">
              <a
                id="dashboard-parent"
                className="collection-item teal"
              >
                <strong>
                   Dashboard
                </strong>
              </a>
              <br />
              <div id="myTab" className="tabs-vertical">
                <ul className="tabs teal">
                  <li className="tab">
                    <Link
                      to="/dashboard/profile"
                      className="white-text"
                    >
                      <i
                        className="fa fa-vcard-o white-text"
                      /> My Profile
                    </Link>
                  </li>
                  <li className="tab">
                    <Link
                      to="/dashboard/recipes"
                      className="white-text"
                    >
                      <i
                        className="fa fa-briefcase white-text"
                      /> My Recipes
                    </Link>
                  </li>
                  <li className="tab">
                    <Link
                      to="/dashboard/favorites"
                      className="white-text"
                    >
                      <i
                        className="fa fa-heart white-text"
                      /> Favorite Recipes
                    </Link>
                  </li>
                  <li className="tab">
                    <Link
                      to="/dashboard/notifications"
                      className="white-text"
                    >
                      <i
                        className="fa fa-comments-o white-text"
                      /> Notifications
                      <span
                        id="not-badge"
                        className="red new badge"
                      >10</span>
                    </Link>
                  </li>
                </ul>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
);

SideNavDashboard.defaultProps = {
  user: {}
};

SideNavDashboard.propTypes = {
  user: PropTypes.shape({
    username: PropTypes.string
  }),
};

export default SideNavDashboard;
