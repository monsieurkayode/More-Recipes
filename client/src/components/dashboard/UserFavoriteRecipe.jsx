import { isEmpty, has } from 'lodash';
import React, { Component } from 'react';
import PropTypes from 'proptypes';
import ReactPaginate from 'react-paginate';

import { Loader } from '../main';
import { WelcomeDisplay } from './';
import { RecipeCardSmall } from '../recipes';

/**
 * @summary - UserFavoriteRecipe class declaration
 * @class UserFavoriteRecipe
 * @extends {Component}
 */
class UserFavoriteRecipe extends Component {
  /**
   * Component constructor
   * @param {object} props
   * @memberOf UserFavoriteRecipe
   */
  constructor(props) {
    super(props);
    this.setPagination = this.setPagination.bind(this);
    this.handlePageClick = this.handlePageClick.bind(this);
  }
  /**
   * @method componentWillReceiveProps
   *
   * @param {object} nextProps
   *
   * @returns {*} any
   */
  componentWillReceiveProps(nextProps) {
    const { recipes } = this.props.userFavorites;
    const currentPageSize = Object.keys(recipes || {}).length;
    const nextPageSize = Object.keys(
      nextProps.userFavorites.recipes || {}
    ).length;
    if (currentPageSize !== nextPageSize) {
      const currentPage = localStorage.getItem('currentPageUserFavorites');
      const nextFavorites = this.hasFavorites() &&
        nextProps.userFavorites.recipes;
      if (!Object.keys(nextFavorites || {}).length && currentPage > 1) {
        localStorage.setItem('currentPageUserFavorites', currentPage - 1);
        this.props.isFetching(true, 'UserFavorites');
        return this.props.fetchUserFavorites(currentPage - 1);
      }
      this.props.fetchUserFavorites(currentPage);
    }
  }

  /**
   * @method setPagination
   *
   * @param {void} void
   *
   * @returns {object} pagination props
   */
  setPagination() {
    if (this.hasFavorites()) {
      const { pagination: { page, pageCount } } = this.props.userFavorites;
      return {
        page,
        pageCount
      };
    }
    return {
      page: 1,
      pageCount: 1
    };
  }

  /**
   * Check if there is a favorite
   * @method hasFavorites
   *
   * @param {void} void
   *
   * @returns {boolean} boolean
   */
  hasFavorites = () => {
    const { userFavorites } = this.props;
    if (isEmpty(userFavorites)) {
      return false;
    }
    if (!isEmpty(userFavorites) && has(userFavorites, 'recipes')) {
      if (!userFavorites.pagination.totalCount) {
        return false;
      }
      return true;
    }
    return true;
  }

  /**
   * Handle page selection for pagination
   * @method handlePageClick
   *
   * @param {object} data
   *
   * @returns {void}
   */
  handlePageClick({ selected }) {
    const page = selected + 1;
    localStorage.setItem('currentPageUserFavorites', page);
    const currentPage = localStorage.getItem('currentPageUserFavorites');
    this.props.isFetching(true, 'UserFavorites');
    this.props.fetchUserFavorites(currentPage);
  }

  /**
   * @method renderUserFavorites
   *
   * @param {number} index
   *
   * @returns {JSX} JSX
   */
  renderUserFavorites = (index) => {
    const recipe = this.props.userFavorites.recipes[index];
    const { removeFavorite, selectRecipe } = this.props;
    return (
      <RecipeCardSmall
        key={recipe.id}
        index={recipe.id}
        recipe={recipe.Recipe}
        handleAction={removeFavorite}
        selectRecipe={selectRecipe}
      />
    );
  }

  /**
   * @method renderPagination
   *
   * @param {void} void
   *
   * @returns {JSX} JSX
   */
  renderPagination = () => {
    const { page, pageCount } = this.setPagination();
    return (
      <div className="pagination-container">
        <ReactPaginate
          previousLabel={<i className="fa fa-chevron-left" />}
          nextLabel={<i className="fa fa-chevron-right" />}
          breakLabel={<a href="">...</a>}
          breakClassName={'break-me'}
          pageCount={pageCount}
          initialPage={page - 1}
          marginPagesDisplayed={2}
          pageRangeDisplayed={5}
          onPageChange={this.handlePageClick}
          disableInitialCallback
          containerClassName={'pagination'}
          subContainerClassName={'pages pagination'}
          activeClassName={'active'}
        />
      </div>
    );
  }
  /**
   * Renders the component
   * @method render
   *
   * @returns {JSX} JSX
   */
  render() {
    const { recipes } = this.props.userFavorites;
    return (
      <div id="favorite-recipes" className="col l9 m12 s12 offset-l3">
        <WelcomeDisplay selected={this.props.selected} />
        <div id="user-favorites" className="row">
          <div className="row">
            {this.hasFavorites() ?
              <div>
                { this.props.isLoadingFavorites ? <Loader /> :
                  Object
                    .keys(recipes)
                    .sort((a, b) => b - a)
                    .map(index => this.renderUserFavorites(index))
                }
              </div> :
              <div className="center-align not-found">
                <img
                  src="/css/img/sad.png"
                  alt=""
                />
                <h5>
                  You have not added any favorites.
                </h5>
              </div>}
          </div>
        </div>
        {this.hasFavorites() && this.renderPagination()}
      </div>
    );
  }
}

UserFavoriteRecipe.defaultProps = {
  userFavorites: {},
  removeFavorite: null
};

UserFavoriteRecipe.propTypes = {
  userFavorites: PropTypes.shape({
    recipes: PropTypes.shape({
      id: PropTypes.number,
      views: PropTypes.number,
      upvote: PropTypes.number,
      downvote: PropTypes.number,
      recipeName: PropTypes.string,
      category: PropTypes.string,
      ingredients: PropTypes.string,
      instructions: PropTypes.string,
      image: PropTypes.string
    }),
    pagination: PropTypes.shape({
      page: PropTypes.number,
      pageCount: PropTypes.number,
      pageSize: PropTypes.number,
      totalCount: PropTypes.number
    })
  }),
  removeFavorite: PropTypes.func,
  selected: PropTypes.string.isRequired,
  selectRecipe: PropTypes.func.isRequired,
  fetchUserFavorites: PropTypes.func.isRequired,
  isFetching: PropTypes.func.isRequired,
  isLoadingFavorites: PropTypes.bool.isRequired
};

export default UserFavoriteRecipe;
