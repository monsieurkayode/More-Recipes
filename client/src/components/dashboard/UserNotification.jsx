import React, { Component } from 'react';
import PropTypes from 'proptypes';
import khaleesi from '../../../assets/css/img/khaleesi.jpeg';

/**
 * @summary - UserNotification class declaration
 * @class UserNotification
 * @extends {Component}
 */
class UserNotification extends Component {
  /**
   * @method componentDidMount
   * @memberof UserNotification
   *
   * @returns {void}
   */
  componentDidMount() {
    const { isFetching } = this.props;
    isFetching(false, 'DashboardPage');
  }

  /**
   * Renders the component
   * @method render
   *
   * @returns {JSX} JSX
   */
  render() {
    return (
      <div
        style={{ marginTop: '10%', fontFamily: 'Roboto, sans-serif' }}
        className="col l9 m12 s12 offset-l3 center-align"
      >
        <img style={{ borderRadius: '50%' }} src={khaleesi} alt="" />
        <div>Khaleesi will notify you once the night king arrives :-)</div>
      </div>
    );
  }
}

UserNotification.propTypes = {
  isFetching: PropTypes.func.isRequired,
};

export default UserNotification;
