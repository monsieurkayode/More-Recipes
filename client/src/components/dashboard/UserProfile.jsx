import React, { Component } from 'react';
import PropTypes from 'proptypes';
import { connect } from 'react-redux';
import isEmpty from 'lodash/isEmpty';

import validateProfileUpdate from '../../utils/validateProfileUpdate';
import { fetchUserDetailsAction, updateProfileAction } from '../../actions';
import holderProfile from '../../../assets/css/img/holder-profile.png';
import { WelcomeDisplay } from './';
import { Loader } from '../main';

/**
 * @summary - UserProfile class declaration
 * @class UserProfile
 * @extends {Component}
 */
class UserProfile extends Component {
  /**
   * Component constructor
   * @param {object} props
   * @memberOf UserProfile
   */
  constructor(props) {
    super(props);
    this.state = {
      bio: '',
      firstName: '',
      lastName: '',
      imageUrl: '',
      imagePreviewUrl: '',
      errors: {},
      disableForm: true
    };
  }
  /**
   * @method componentDidMount
   *
   * @returns {undefined}
   */
  componentDidMount() {
    this.props.fetchUserDetailsAction();
  }

  /**
   * @method componentWillReceiveProps
   *
   * @param {object} nextProps
   *
   * @returns {undefined}
   */
  componentWillReceiveProps(nextProps) {
    if (!isEmpty(nextProps.userDetails)) {
      const {
        bio,
        lastName,
        firstName,
        imageUrl
      } = nextProps.userDetails;
      this.setState({
        bio: bio || '',
        firstName: firstName || '',
        lastName: lastName || '',
        imageUrl: imageUrl || ''
      });
    }
  }

  /**
   * Handle client-side input validation
   * @method isValid
   *
   * @returns {undefined}
   */
  isValid() {
    const { errors, isValid } = validateProfileUpdate(this.state);
    if (!isValid) {
      this.setState({ errors });
    }
    return isValid;
  }

  /**
   * Handle input change
   * @method handleInputChange
   *
   * @param {object} event
   *
   * @returns {undefined}
   */
  handleInputChange = (event) => {
    this.setState({
      [event.target.name]: event.target.value
    });
  }

  /**
   * Handle image selection change
   * @method handleImageChange
   *
   * @param {object} event
   *
   * @returns {undefined}
   */
  handleImageChange = (event) => {
    event.preventDefault();

    const reader = new FileReader();
    const file = event.target.files[0];

    reader.onloadend = () => {
      this.setState({
        imageUrl: file,
        imagePreviewUrl: reader.result
      });
    };
    reader.readAsDataURL(file);
  }

  /**
   * Reset form to initial state
   * @method resetForm
   *
   * @param {object} event
   *
   * @returns {undefined}
   */
  resetForm = (event) => {
    event.preventDefault();
    const { bio, firstName, lastName, imageUrl } = this.props.userDetails;
    this.setState({
      disableForm: true,
      bio,
      firstName,
      lastName,
      imageUrl,
      imagePreviewUrl: '',
      errors: {}
    });
  }

  /**
   * Disable or enable form inputs
   * @method toggleForm
   *
   * @param {object} event
   *
   * @returns {undefined}
   */
  toggleForm = (event) => {
    event.preventDefault();
    this.setState({
      disableForm: !this.state.disableForm
    });
  }

  /**
   * Hanlde Submit
   * @method handleSubmit
   *
   * @param {object} event
   *
   * @returns {undefined}
   */
  handleSubmit = (event) => {
    event.preventDefault();
    const userDetails = { ...this.state };
    if (this.isValid()) {
      this.setState({ errors: {} });
      this.props.updateProfileAction(userDetails)
        .then(() => this.setState({ disableForm: true }));
    }
  }

  /**
   * Renders the component
   * @method render
   *
   * @returns {JSX} JSX
   */
  render() {
    const { imagePreviewUrl } = this.state;
    let imagePreview = null;
    if (imagePreviewUrl) {
      imagePreview = (<img src={imagePreviewUrl} alt="" />);
    } else {
      imagePreview = (<img
        className="responsive-img"
        src={this.props.userDetails.imageUrl || holderProfile}
        alt=""
      />);
    }
    const { username } = this.props.user;
    const { firstName, lastName } = this.state.errors;
    return (
      <div id="user-profile" className="col l9 m12 s12 offset-l3">
        <WelcomeDisplay selected={this.props.selected} />
        <div id="my-profile" className="row">
          { this.props.isLoadingProfile && <Loader /> }
          { !this.props.isLoadingProfile &&
          <div className="col l12 m12 s12">
            <div className="row">
              <div className="col l4 m4">
                <div className="card z-depth-2">
                  <div className="card-image">
                    {imagePreview}
                  </div>
                </div>
              </div>
              <div className="col l8 m8 s12">
                <form id="profile-form" onSubmit={this.handleSubmit}>
                  <div className="browser-default">
                    <label htmlFor="firstName">First Name</label>
                    <input
                      type="text"
                      name="firstName"
                      onChange={this.handleInputChange}
                      value={this.state.firstName}
                      disabled={this.state.disableForm}
                    />
                    {firstName && <span className="red-text">{firstName}</span>}
                  </div>
                  <div className="browser-default">
                    <label htmlFor="lastName">Last Name</label>
                    <input
                      type="text"
                      name="lastName"
                      onChange={this.handleInputChange}
                      value={this.state.lastName}
                      disabled={this.state.disableForm}
                    />
                    {lastName && <span className="red-text">{lastName}</span>}
                  </div>
                  <div className="browser-default">
                    <label htmlFor="handle">Public Display Name</label>
                    <input
                      type="text"
                      defaultValue={`@${username}`}
                      disabled
                    />
                  </div>
                  <div className="browser-default">
                    <label htmlFor="about-me">About Me</label>
                    <textarea
                      name="bio"
                      value={this.state.bio}
                      spellCheck={false}
                      onChange={this.handleInputChange}
                      disabled={this.state.disableForm}
                    />
                  </div>
                  {!this.state.disableForm &&
                  <div id="profile-image-btn">
                    <label htmlFor="imageUrl">
                      <input
                        onChange={this.handleImageChange}
                        type="file"
                        id="imageUrl"
                        name="imageUrl"
                      />
                      <span
                        className="btn"
                      >
                      Upload Image
                      </span>
                    </label>
                  </div>}
                  {this.state.disableForm ?
                    <button
                      onClick={this.toggleForm}
                      className="btn right grey darken-1 profile-edit-btn"
                    >
                      <i className="fa fa-pencil" /> Edit Profile
                    </button> :
                    <div className="profile-btn-group">
                      <button
                        onClick={this.resetForm}
                        className="btn grey darken-1 right"
                      >
                        <i className="fa fa-times" /> Discard
                      </button>
                      <button
                        type="submit"
                        onClick={this.handleSubmit}
                        className="btn right grey darken-1"
                      >
                        <i className="fa fa-save" /> Save Profile
                      </button>
                    </div>
                  }
                </form>
              </div>
            </div>
          </div>}
        </div>
      </div>
    );
  }
}

UserProfile.defaultProps = {
  userDetails: {}
};

UserProfile.propTypes = {
  user: PropTypes.shape({
    username: PropTypes.string.isRequired
  }).isRequired,
  selected: PropTypes.string.isRequired,
  fetchUserDetailsAction: PropTypes.func.isRequired,
  updateProfileAction: PropTypes.func.isRequired,
  userDetails: PropTypes.shape({
    bio: PropTypes.string,
    firstName: PropTypes.string,
    lastName: PropTypes.string,
    imageUrl: PropTypes.string,
  }),
  isLoadingProfile: PropTypes.bool.isRequired
};

const mapStateToProps = ({ userDetails }) => ({ userDetails });

export default connect(mapStateToProps, {
  fetchUserDetailsAction,
  updateProfileAction
})(UserProfile);

