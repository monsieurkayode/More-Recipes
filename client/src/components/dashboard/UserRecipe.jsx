import { isEmpty, has } from 'lodash';
import React, { Component } from 'react';
import PropTypes from 'proptypes';
import ReactPaginate from 'react-paginate';

import { RecipeCardSmall } from '../recipes';
import { WelcomeDisplay } from './';
import { Loader } from '../main';

/**
 * @summary - UserRecipe class declaration
 * @class UserRecipe
 * @extends {Component}
 */
class UserRecipe extends Component {
  /**
   * Component constructor
   * @param {object} props
   * @memberOf UserRecipe
   */
  constructor(props) {
    super(props);
    this.setPagination = this.setPagination.bind(this);
    this.handlePageClick = this.handlePageClick.bind(this);
  }
  /**
   * @method componentWillReceiveProps
   *
   * @param {object} nextProps
   *
   * @returns {*} any
   */
  componentWillReceiveProps(nextProps) {
    const { recipes } = this.props.userRecipes;
    const currentPageSize = Object.keys(recipes || {}).length;
    const nextPageSize = Object.keys(
      nextProps.userRecipes.recipes || {}
    ).length;
    if (currentPageSize !== nextPageSize) {
      const currentPage = localStorage.getItem('currentPageUserRecipes');
      const nextRecipes = this.hasRecipes() && nextProps.userRecipes.recipes;
      if (!Object.keys(nextRecipes || {}).length && currentPage > 1) {
        localStorage.setItem('currentPageUserRecipes', currentPage - 1);
        this.props.isFetching(true, 'UserRecipes');
        return this.props.fetchUserRecipes(currentPage - 1);
      }
      this.props.fetchUserRecipes(currentPage);
    }
  }

  /**
   * @method setPagination
   *
   * @param {void} void
   *
   * @returns {object} pagination props
   */
  setPagination() {
    if (this.hasRecipes()) {
      const { pagination: { page, pageCount } } = this.props.userRecipes;
      return {
        page,
        pageCount
      };
    }
    return {
      page: 1,
      pageCount: 1
    };
  }

  /**
   * Check if there is a recipe
   * @method hasFavorites
   *
   * @param {void} void
   *
   * @returns {boolean} boolean
   */
  hasRecipes = () => {
    const { userRecipes } = this.props;
    if (isEmpty(userRecipes)) {
      return false;
    }
    if (!isEmpty(userRecipes) && has(userRecipes, 'recipes')) {
      if (userRecipes.pagination.totalCount === 0) {
        return false;
      }
      return true;
    }
    return true;
  }


  /**
   * Handle page selection for pagination
   * @method handlePageClick
   *
   * @param {object} data
   *
   * @returns {void}
   */
  handlePageClick({ selected }) {
    const page = selected + 1;
    localStorage.setItem('currentPageUserRecipes', page);
    const currentPage = localStorage.getItem('currentPageUserRecipes');
    this.props.isFetching(true, 'UserRecipes');
    this.props.fetchUserRecipes(currentPage, this);
  }

  /**
   * @method renderUserRecipes
   *
   * @param {number} index
   *
   * @returns {JSX} JSX
   */
  renderUserRecipes = (index) => {
    const recipe = this.props.userRecipes.recipes[index];
    const { selected, selectRecipe } = this.props;
    return (
      <RecipeCardSmall
        key={recipe.id}
        index={recipe.id}
        recipe={recipe}
        selected={selected}
        selectRecipe={selectRecipe}
      />
    );
  }

  /**
   * @method renderPagination
   *
   * @param {void} void
   *
   * @returns {JSX} JSX
   */
  renderPagination = () => {
    const { page, pageCount } = this.setPagination();
    return (
      <div className="pagination-container">
        <ReactPaginate
          previousLabel={<i className="fa fa-chevron-left" />}
          nextLabel={<i className="fa fa-chevron-right" />}
          breakLabel={<a href="">...</a>}
          breakClassName={'break-me'}
          pageCount={pageCount}
          initialPage={page - 1}
          marginPagesDisplayed={2}
          pageRangeDisplayed={5}
          onPageChange={this.handlePageClick}
          disableInitialCallback
          containerClassName={'pagination'}
          subContainerClassName={'pages pagination'}
          activeClassName={'active'}
        />
      </div>
    );
  }

  /**
   * Renders the component
   * @method render
   *
   * @returns {JSX} JSX
   */
  render() {
    const { recipes } = this.props.userRecipes;
    return (
      <div id="user-recipes" className="col l9 m12 s12 offset-l3">
        <WelcomeDisplay selected={this.props.selected} />
        <div id="my-recipes" className="row">
          {this.hasRecipes() ?
            <div>
              { this.props.isLoadingRecipes ? <Loader /> :
                Object
                  .keys(recipes)
                  .sort((a, b) => b - a)
                  .map(index => this.renderUserRecipes(index))
              }
            </div> :
            <div className="center-align not-found">
              <img
                src="/css/img/sad.png"
                alt=""
              />
              <h5>
                There is nothing here, create and share awesome recipes.
              </h5>
            </div>}
        </div>
        {this.hasRecipes() && this.renderPagination()}
      </div>
    );
  }
}

UserRecipe.defaultProps = {
  userRecipes: {},
  deletePost: null
};

UserRecipe.propTypes = {
  userRecipes: PropTypes.shape({
    recipes: PropTypes.shape({
      id: PropTypes.number,
      views: PropTypes.number,
      upvote: PropTypes.number,
      downvote: PropTypes.number,
      recipeName: PropTypes.string,
      category: PropTypes.string,
      ingredients: PropTypes.string,
      instructions: PropTypes.string,
      image: PropTypes.string
    }),
    pagination: PropTypes.shape({
      page: PropTypes.number,
      pageCount: PropTypes.number,
      pageSize: PropTypes.number,
      totalCount: PropTypes.number
    })
  }),
  selected: PropTypes.string.isRequired,
  isFetching: PropTypes.func.isRequired,
  selectRecipe: PropTypes.func.isRequired,
  fetchUserRecipes: PropTypes.func.isRequired,
  isLoadingRecipes: PropTypes.bool.isRequired,
};

export default UserRecipe;
