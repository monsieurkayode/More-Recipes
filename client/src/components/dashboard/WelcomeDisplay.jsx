import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'proptypes';

/**
 * WelcomeDisplay
 * @function WelcomeDisplay
 *
 * @param {object} props
 *
 * @return {JSX} JSX
 */
const WelcomeDisplay = (props) => {
  const { user: { username }, selected } = props;
  return (
    <div id="welcome-panel" className="card-panel grey lighten-4">
      <span className="card-content">
        <h5>Welcome {username}!</h5>
        <span>Dashboard / My {selected}</span>
      </span>
    </div>
  );
};

WelcomeDisplay.defaultProps = {
  user: {}
};

WelcomeDisplay.propTypes = {
  user: PropTypes.shape({
    username: PropTypes.string
  }),
  selected: PropTypes.string.isRequired
};

const mapStateToProps = ({ signinState }) => ({
  user: signinState.user,
});


export default connect(mapStateToProps)(WelcomeDisplay);
