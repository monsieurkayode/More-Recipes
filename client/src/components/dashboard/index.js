/* jshint esversion: 6 */
import UserProfile from './UserProfile';
import UserRecipe from './UserRecipe';
import WelcomeDisplay from './WelcomeDisplay';
import DashboardPanel from './DashboardPanel';
import SideNavDashboard from './SideNavDashboard';
import UserFavoriteRecipe from './UserFavoriteRecipe';
import UserNotification from './UserNotification';

export {
  UserProfile,
  UserRecipe,
  UserFavoriteRecipe,
  UserNotification,
  WelcomeDisplay,
  SideNavDashboard,
  DashboardPanel
};
