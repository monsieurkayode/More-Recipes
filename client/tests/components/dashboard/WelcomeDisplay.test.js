/* eslint-disable no-undef */
import React from 'react';
import { Provider } from 'react-redux';

import { WelcomeDisplay } from '../../../src/components/dashboard';


const setup = (username, selected) => {
  const store = mockStore({
    signinState: {
      user: {
        username
      }
    }
  });
  const props = {
    user: store.getState().user,
    selected
  };
  const wrapper = mount(
    <Provider store={store}>
      <WelcomeDisplay {...props} />
    </Provider>
  );
  return {
    wrapper
  };
};

describe('The WelcomeDisplay Component', () => {
  it('should render correctly', (done) => {
    const { wrapper } = setup('username', 'recipes');
    expect(toJson(wrapper)).toMatchSnapshot();
    done();
  });
});
