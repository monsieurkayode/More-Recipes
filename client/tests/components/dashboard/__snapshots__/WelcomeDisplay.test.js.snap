// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The WelcomeDisplay Component should render correctly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(WelcomeDisplay)
    selected="recipes"
  >
    <WelcomeDisplay
      dispatch={[Function]}
      selected="recipes"
      user={
        Object {
          "username": "username",
        }
      }
    >
      <div
        className="card-panel grey lighten-4"
        id="welcome-panel"
      >
        <span
          className="card-content"
        >
          <h5>
            Welcome 
            username
            !
          </h5>
          <span>
            Dashboard / My 
            recipes
          </span>
        </span>
      </div>
    </WelcomeDisplay>
  </Connect(WelcomeDisplay)>
</Provider>
`;
